src/gmock-win32.cpp

```diff
@@ -167,7 +186,9 @@ namespace utils {
             DWORD oldProtect{ };
             if (core->pfn_VirtualProtect(address, size, PAGE_WRITECOPY, &oldProtect))
             {               
-                std::shared_ptr< void > finalAction(nullptr, [&](auto&&...) {
+                std::shared_ptr< void > finalAction(nullptr, [&](auto&&...)
+                {
+                    core->pfn_FlushInstructionCache(processHandle, address, size);
                     core->pfn_VirtualProtect(address, size, oldProtect, &oldProtect);
```

Comment:
> Mmm... `FlushInstructionCache` Really like :)

I found some code where `FlushInstructionCache` was used. It is recommended to use this function after making memory changes, before executing the code.

`When protecting a region that will be executable, the calling program bears responsibility for ensuring cache coherency via an appropriate call to FlushInstructionCache once the code has been set in place. Otherwise attempts to execute code out of the newly executable region may produce unpredictable results.`

So, I added it just in case )